packages/build-tools/src/utils/hooks.ts

```diff
@@ -30,7 +31,7 @@ export async function runHookIfPresent<TJob extends Job>(
     ctx.logger.warn(`Failed to parse or read package.json: ${err.message}`);
     return;
   }
-  if (packageJson.scripts?.[hook]) {
+  if (packageJson!.scripts?.[hook]) {
```

Comment:
When `readPackageJson` was still a local function, it returned `Promise<PackageJson>` because that is specifically what hooks needed, now it returns `Promise<any>` because `JSON.parse()` returns `any`, and it's warning me that it could be `undefined` (which it can't, functionally, because we have the error handling in that function for that).

Reply:
I see. 
Just an idea: maybe it's worth using https://github.com/sindresorhus/type-fest There's a `PackageJSON` type

Reply:
If we would need anything else from it, then yes. With the existing interface just for reading the hook scripts (almost a one-liner), and this new feature that just logs the entire thing (only parsing it for pretty-printing), I don't think it warrants adding an external dependency for it, or pulling in a 650+ line type definition.

Reply:
in general consider using assert instead of `!`

but in this case, I think you can just change 
`let packageJson: PackageJson | undefined;` to `let packageJson: PackageJson;`

not tested